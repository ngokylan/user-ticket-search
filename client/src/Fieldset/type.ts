import { ReactNode } from 'react';

export type FieldsetProps = {
  /** Fieldset Enable/Disable status, (optional) */
  isDisabled?: boolean;
  /** Fieldset id attribute, optional id */
  id?: string;
  /** Fieldset class name attribute, (optional) */
  className?: string;
  /** Fieldset content */
  children: ReactNode;
};
