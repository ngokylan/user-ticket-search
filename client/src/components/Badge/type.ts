import { ReactNode } from 'react';

export type BadgeType =
  | 'grey'
  | 'outlined'
  | 'warning'
  | 'info'
  | 'success'
  | 'danger';
export const BadgeTypes: { [key: string]: BadgeType } = {
  grey: 'grey',
  outlined: 'outlined',
  warning: 'warning',
  info: 'info',
  success: 'success',
  danger: 'danger',
};

export type BadgeRole = 'info' | 'status' | 'label';

export type BadgeProps = {
  /**
   * Badge type, use one from the following:
   * BadgeType = 'grey' | 'outlined' | 'warning' | 'info' | 'success' | 'danger'
   */
  type: BadgeType;
  /**
   * Badge role, use one from the following:
   * BadgeRole = 'info' | 'status' | 'label'
   */
  role?: BadgeRole;
  /** Custom id attribute */
  id?: string;
  /** Add custom class to class attribute */
  className?: string;
  /** To enable popover Icon */
  hasPopoverIcon?: boolean;
  /** Badge icon */
  icon?: ReactNode;
  /** The content of a badge */
  children?: ReactNode;
  /** To render badge in circular */
  isCircular?: boolean;
};
