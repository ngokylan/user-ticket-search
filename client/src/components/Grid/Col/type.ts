import { ReactNode } from 'react';

export type ColResponsiveOption = number | object;
export type ColAlign =
  | null
  | 'auto'
  | 'start'
  | 'end'
  | 'center'
  | 'baseline'
  | 'stretch';

export type ColProps = {
  /** Content of col */
  children?: ReactNode;
  /** Span atribute of col */
  span?: number;
  /** If is auto */
  isAuto?: boolean;
  /** order atribute of col */
  order?: number;
  /** offset atribute of col */
  offset?: number;
  /**
   * align atribute of col, use one from the following:
   * ColAlign =  null | 'auto' | 'start' | 'end' | 'center' | 'baseline' | 'stretch'
   */
  align?: ColAlign;
  /** shrink atribute of col */
  shrink?: number;
  /** grow atribute of col */
  grow?: number;
  /** Fill atribute of col */
  isFill?: boolean;
  /** xs responsive atribute of col */
  xs?: ColResponsiveOption;
  /** sm responsive atribute of col */
  sm?: ColResponsiveOption;
  /** md responsive atribute of col */
  md?: ColResponsiveOption;
  /** lg responsive atribute of col */
  lg?: ColResponsiveOption;
  /** xl responsive atribute of col */
  xl?: ColResponsiveOption;
  /** xxl responsive atribute of col */
  xxl?: ColResponsiveOption;
  /** xxl responsive atribute of col */
  xxxl?: ColResponsiveOption;
  /** key atribute of col */
  key?: number | string;
  /** Col class name attribute, (optional) */
  className?: string;
  /** WCAG role */
  role?: string;
};

export type ColResponsivePropertiesType = {
  property: string;
  prefix: string;
};
