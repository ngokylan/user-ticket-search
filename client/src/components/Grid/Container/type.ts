import { ReactNode } from 'react';
export type ContainerSize = 'lg';

export type ContainerProps = {
  /** Content of container */
  children?: ReactNode;
  /** isFluid atribute of container */
  isFluid?: boolean;
  /** size atribute of container */
  size?: ContainerSize;
  /** Role is for WCAG accessibility compliance (optional) */
  role?: string;
};
