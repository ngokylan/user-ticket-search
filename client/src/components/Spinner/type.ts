export type SpinnerSize = 'xl' | 'l' | 'm' | 's';

export type SpinnerProps = {
  color?: string;
  /** Custom id attribute */
  id?: string;
  size?: SpinnerSize;
  /** Role is for WCAG accessibility compliance, defaults to "status" if omitted, (optional) */
  role?: string;
  /** Aria Label is for WCAG accessibility compliance, defaults to "Loading..." if omitted, (optional) */
  ariaLabel?: string;
};
