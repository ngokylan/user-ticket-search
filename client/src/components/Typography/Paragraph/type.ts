import { ReactNode } from 'react';
import {
  TypographyAlign,
  TypographyDecoration,
  TypographyTransform,
  TypographyColor,
} from '../type';
export type ParagraphSize = 'lg' | 'rg' | 'sm' | 'xs';

export type ParagraphProps = {
  /** ID of the component */
  id?: string;
  /** Children */
  children?: ReactNode;
  /** Size that is defined in the  Design System : 'lg', 'rg', 'sm', 'xs' */
  size?: ParagraphSize;
  /** It aligns its node : 'left', 'center', 'right', 'justify' */
  align?: TypographyAlign;
  /** It decorates its text node : 'overline', 'lineThrough', 'underline' */
  decoration?: TypographyDecoration;
  /** Style the text to be bold */
  isBold?: boolean;
  /** Style the text to be italic */
  isItalic?: boolean;
  /** Truncate the text in narrow space */
  isTruncate?: boolean;
  /** Style the text node to transform into : 'lowercase', 'uppercase', 'capitalize' */
  transform?: TypographyTransform;
  /** Style the text node's colour into one of defined in  Design System :
   * 'primary', 'secondary', 'black', 'information', 'warning', 'success', 'danger'
   * To ensure WCAG compliance, you must ensure that text color has sufficient
   * color contrast - for more information, see https://www.w3.org/TR/WCAG21/#contrast-minimum
   */
  color?: TypographyColor;
  /** class name attribute, (optional) */
  className?: string;
};
