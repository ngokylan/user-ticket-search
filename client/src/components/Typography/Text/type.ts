import { ReactNode } from 'react';
import {
  TypographyDecoration,
  TypographyTransform,
  TypographyColor,
} from '../type';
export type TextHtmlTag = 'span' | 'strong' | 'small';
export type TextSize = 'lg' | 'rg' | 'sm' | 'xs' | 'button' | 'status';

export type TextProps = {
  /** ID of the component */
  id?: string;
  /** Children */
  children?: ReactNode;
  /** HTML tag to render the component as : 'span' | 'strong' | 'small' */
  htmlTag?: TextHtmlTag;
  /** Size that is defined in the Design System : 'lg' | 'rg' | 'sm' | 'xs' | 'button' | 'status' */
  size?: TextSize;
  /** It decorates its text node : 'overline', 'lineThrough', 'underline' */
  decoration?: TypographyDecoration;
  /** Style the text to be bold */
  isBold?: boolean;
  /** Style the text to be italic */
  isItalic?: boolean;
  /** Style the text node to transform into : 'lowercase', 'uppercase', 'capitalize' */
  transform?: TypographyTransform;
  /** Style the text node's colour into one of defined in Design System :
   * 'primary', 'secondary', 'black', 'gray', 'information', 'warning', 'success', 'danger'
   */
  color?: TypographyColor;
  /** class name attribute, (optional) */
  className?: string;
};
