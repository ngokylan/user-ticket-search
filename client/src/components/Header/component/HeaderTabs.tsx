import React from 'react';
import './HeaderTabs.scss';
import { HeaderTabsProps } from '../type';
import { getClass } from '../../../_lib/helper';

function HeaderTabs({ tabs }: HeaderTabsProps) {
  return (
    <div className="tabs" id="tab">
      <nav className="tabs__nav" id="tab-nav">
        <ul>
          {Array.isArray(tabs) &&
            tabs.map((item, index) => (
              <li
                className={getClass('tabs__item', '', {
                  active: item.active,
                })}
                key={index}
              >
                <a onClick={item.onClick}>{item.label}</a>
              </li>
            ))}
        </ul>
      </nav>
    </div>
  );
}

export default HeaderTabs;
