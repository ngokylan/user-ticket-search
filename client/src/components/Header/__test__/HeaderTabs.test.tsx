import React from 'react';
import { mount } from 'enzyme';
import HeaderTabs from '../component/HeaderTabs';
import { HeaderTabsProps } from '../type';

const defaultProps: HeaderTabsProps = undefined;
const mountComponent = (customProps: any = {}) => {
  const props = { ...defaultProps, ...customProps };
  const wrapper = mount(<HeaderTabs {...props} />);

  return { props, wrapper };
};

describe('Header::HeaderTabs', () => {
  describe('rendering', () => {
    it('should render component without any crash', () => {
      const { wrapper } = mountComponent();
      expect(wrapper).toHaveLength(1);
    });

    it('should be match with snapshot', () => {
      const { wrapper } = mountComponent();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('props', () => {
    describe('default', () => {
      it('should have expected default props', () => {
        expect(JSON.stringify(defaultProps)).toBe(
          JSON.stringify(HeaderTabs.defaultProps)
        );
      });
    });

    describe('custom', () => {
      it('should render list item for tabs prop', () => {
        jest.clearAllMocks();
        const tabObject = {
          active: true,
          onClick: jest.fn(),
          label: 'A',
        };

        const { wrapper } = mountComponent({
          tabs: [tabObject],
        });

        const liElem = wrapper.find('li');
        expect(liElem.prop('className')).toBe(
          'tabs__item tabs__item--active'
        );
        const aElem = liElem.find('a');
        expect(aElem.prop('onClick')).toBe(tabObject.onClick);
        expect(aElem.prop('children')).toBe(tabObject.label);
      });
    });
  });
});
