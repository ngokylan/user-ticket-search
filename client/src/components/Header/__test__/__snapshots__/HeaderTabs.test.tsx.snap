// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header::HeaderTabs rendering should be match with snapshot 1`] = `
<div
  className="tabs"
  id="tab"
>
  <nav
    className="tabs__nav"
    id="tab-nav"
  >
    <ul />
  </nav>
</div>
`;
